content/src/debug.js

```diff
@@ -2,47 +2,42 @@
 // DEBUGGER SUPPORT
 ///////////////////////////////////////////////////////////////////////////
 
+
 define([
   'jquery',
   'view',
   'see',
-  'sourcemap/source-map-consumer'],
+  'sourcemap/source-map-consumer'
+ ],
 function($, view, see, sourcemap) {
 
 eval(see.scope('debug'));
 
-var targetWindow = null;    // window object of the frame being debugged.
-var nextDebugId = 1;        // a never-decreasing sequence of debug event ids.
-var firstSessionId = 1;     // the first id of the current running session.
-var debugIdException = {};  // map debug ids -> exception objects.
-var debugIdRecord = {};     // map debug ids -> line execution records.
-var lineRecord = {};        // map line numbers -> line execution records.
-var everyRecord = [];       // a sequence of all records from the run.
-var cachedSourceMaps = {};  // parsed source maps for currently-running code.
-var cachedParsedStack = {}; // parsed stack traces for currently-running code.
-var pollTimer = null;       // poll for stop button.
-var stopButtonShown = 0;    // 0 = not shown; 1 = shown; 2 = stopped.
-var throwNeeded = !(new Error).stack;
-var lineNumberOffset = 3;   // Lines of Coffee boilerplate above user code.
+var targetWindow = null;      // window object of the frame being debugged.
+var currentEventIndex = 0;    // current index into traceEvents.
+var currentDebugId = 0;       // id used to pair jquery-turtle events with trace events.
+var debugRecordsByDebugId = {}; // map debug ids -> line execution records.
+var debugRecordsByLineNo = {};  // map line numbers -> line execution records.
+var cachedParseStack = {};    // parsed stack traces for currently-running code.
+var pollTimer = null;         // poll for stop button.
+var stopButtonShown = 0;      // 0 = not shown; 1 = shown; 2 = stopped.
+var currentSourceMap = null;  // v3 source map for currently-running instrumented code.
+var traceEvents = [];         // list of event location objects created by tracing events
+
 
 Error.stackTraceLimit = 20;
 
+
 // Resets the debugger state:
 // Remembers the targetWindow, and clears all logged debug records.
 // Calling bindframe also resets firstSessionId, so that callbacks
 // having to do with previous sessions are ignored.
 function bindframe(w) {
-  if (!targetWindow && !w || targetWindow === w) return;
+  if (!targetWindow && !w || targetWindow == w) return;
   targetWindow = w;
```

Comment:
This is the function that is called when a run is starting.

To avoid confusing debug information from a previous run, you should reset all of your debugging data structures here.  For example, `debugRecordsByDebugId = {}`